superset/utils/webdriver.py

```diff
@@ -0,0 +1,132 @@
+
+
+class WebDriverProxy:
+    def __init__(
+        self, driver_type: str, window: Optional[WindowSize] = None,
+    ):
+        self._driver_type = driver_type
+        self._window: WindowSize = window or (800, 600)
+
+    def create(self) -> WebDriver:
+        if self._driver_type == "firefox":
+            driver_class = firefox.webdriver.WebDriver
+            options = firefox.options.Options()
+        elif self._driver_type == "chrome":
+            driver_class = chrome.webdriver.WebDriver
+            options = chrome.options.Options()
+            options.add_argument(f"--window-size={self._window[0]},{self._window[1]}")
+        else:
+            raise Exception(f"Webdriver name ({self._driver_type}) not supported")
+        # Prepare args for the webdriver init
+
+        # Add additional configured options
+        for arg in current_app.config["WEBDRIVER_OPTION_ARGS"]:
+            options.add_argument(arg)
+
+        kwargs: Dict[Any, Any] = dict(options=options)
+        kwargs.update(current_app.config["WEBDRIVER_CONFIGURATION"])
+        logger.info("Init selenium driver")
+
+        return driver_class(**kwargs)
+
+    def auth(self, user: "User") -> WebDriver:
+        driver = self.create()
+        # Setting cookies requires doing a request first
+        driver.get(headless_url("/login/"))
+        return machine_auth_provider_factory.instance.authenticate_webdriver(
+            driver, user
+        )
+
+    @staticmethod
+    def destroy(driver: WebDriver, tries: int = 2) -> None:
+        """Destroy a driver"""
+        # This is some very flaky code in selenium. Hence the retries
+        # and catch-all exceptions
+        try:
+            retry_call(driver.close, tries=tries)
+        except Exception:  # pylint: disable=broad-except
+            pass
+        try:
+            driver.quit()
+        except Exception:  # pylint: disable=broad-except
+            pass
+
+    def get_screenshot(
+        self,
+        url: str,
+        element_name: str,
+        user: "User",
+        retries: int = SELENIUM_RETRIES,
+    ) -> Optional[bytes]:
+        driver = self.auth(user)
+        driver.set_window_size(*self._window)
+        driver.get(url)
+        img: Optional[bytes] = None
+        logger.debug("Sleeping for %i seconds", SELENIUM_HEADSTART)
+        time.sleep(SELENIUM_HEADSTART)
+        try:
+            logger.debug("Wait for the presence of %s", element_name)
+            element = WebDriverWait(
+                driver, current_app.config["SCREENSHOT_LOCATE_WAIT"]
+            ).until(EC.presence_of_element_located((By.CLASS_NAME, element_name)))
+            logger.debug("Wait for .loading to be done")
+            WebDriverWait(driver, current_app.config["SCREENSHOT_LOAD_WAIT"]).until_not(
+                EC.presence_of_all_elements_located((By.CLASS_NAME, "loading"))
+            )
+            logger.info("Taking a PNG screenshot")
```

Comment:
While we're at it, let's put a bit more info here: `logger.info("Taking a PNG screenshot %s", url)`